### ISSUE TITLE
Install ODL from tar/zip

### ISSUE THREAD

**Issue by dfarrell07** [dfarrell07/puppet-opendaylight#41]
**This is not a feature that's meant for general users to consume**.

The devs doing OpenStack+OpenDaylight integration in the context of OPNFV need a way to install a custom built ODL from an archive, to test their changes/fixes.

**Comment by dfarrell07** [comment]
The version of the code that's producing this isn't commited yet (failing tests), FYI.

I'm seeing this error from `camptocamp/puppet-archive` during CentOS 7 Beaker tests.

```
 38) opendaylight class testing Karaf config file overriding default features and passing extra features should work idempotently with no errors
     Failure/Error: apply_manifest(pp, :catch_failures => true)
     Beaker::Host::CommandFailure:
       Host 'centos-7' exited with 4 running:
        puppet apply --verbose --detailed-exitcodes /tmp/apply_manifest.pp.hNFxF9
       Last 10 lines of output were:
       	Notice: /Stage[main]/Opendaylight::Install/Archive[opendaylight-0.2.2]/Archive::Extract[opendaylight-0.2.2]/Exec[opendaylight-0.2.2 unpack]/returns: tar: Unexpected EOF in archive
       	Notice: /Stage[main]/Opendaylight::Install/Archive[opendaylight-0.2.2]/Archive::Extract[opendaylight-0.2.2]/Exec[opendaylight-0.2.2 unpack]/returns: tar: Unexpected EOF in archive
       	Notice: /Stage[main]/Opendaylight::Install/Archive[opendaylight-0.2.2]/Archive::Extract[opendaylight-0.2.2]/Exec[opendaylight-0.2.2 unpack]/returns: tar: Error is not recoverable: exiting now
       	Notice: /Stage[main]/Opendaylight::Config/File[org.apache.karaf.features.cfg]: Dependency Exec[opendaylight-0.2.2 unpack] has failures: true
       	Notice: /Stage[main]/Opendaylight::Service/Service[opendaylight]: Dependency Exec[opendaylight-0.2.2 unpack] has failures: true
       	Error: mkdir -p /opt && tar --no-same-owner --no-same-permissions -xzf /usr/src/opendaylight-0.2.2.tar.gz -C /opt returned 2 instead of one of [0]
       	Error: /Stage[main]/Opendaylight::Install/Archive[opendaylight-0.2.2]/Archive::Extract[opendaylight-0.2.2]/Exec[opendaylight-0.2.2 unpack]/returns: change from notrun to 0 failed: mkdir -p /opt && tar --no-same-owner --no-same-permissions -xzf /usr/src/opendaylight-0.2.2.tar.gz -C /opt returned 2 instead of one of [0]
       	Warning: /Stage[main]/Opendaylight::Config/File[org.apache.karaf.features.cfg]: Skipping because of failed dependencies
       	Warning: /Stage[main]/Opendaylight::Service/Service[opendaylight]: Skipping because of failed dependencies
       	Notice: Finished catalog run in 0.26 seconds
```

**Comment by dfarrell07** [comment]
As of 7d11ebb (last night), most of the functionality associated with this feature has been added. Need to update the docs to reflect the API change, bump the minor version and add additional test coverage before I merge it into master.

**Comment by dfarrell07** [comment]
> Need to update the docs to reflect the API change,

Done

> bump the minor version

Not done, kinda got thrown off when #35 changed the API without a version bump as well. Not a priority at all, as project is still super new and not on the Forge.

> add additional test coverage before I merge it into master.

Working on it, but blocked by #43, which is going to be harder than I thought.

Went ahead and merged into master so folks can consume. Will raise a different issue for test coverage.
